components/usage/pkg/apiv1/usage.go

```diff
@@ -121,6 +122,116 @@ func (s *UsageService) ListBilledUsage(ctx context.Context, in *v1.ListBilledUsa
 	}, nil
 }
 
+func (s *UsageService) ListUsage(ctx context.Context, in *v1.ListUsageRequest) (*v1.ListUsageResponse, error) {
+	to := time.Now()
+	if in.To != nil {
+		to = in.To.AsTime()
+	}
+	from := to.Add(-maxQuerySize)
+	if in.From != nil {
+		from = in.From.AsTime()
+	}
+
+	if from.After(to) {
+		return nil, status.Errorf(codes.InvalidArgument, "Specified From timestamp is after To. Please ensure From is always before To")
+	}
+
+	if to.Sub(from) > maxQuerySize {
+		return nil, status.Errorf(codes.InvalidArgument, "Maximum range exceeded. Range specified can be at most %s", maxQuerySize.String())
+	}
+
+	order := db.DescendingOrder
+	if in.Order == v1.ListUsageRequest_ORDERING_ASCENDING {
+		order = db.AscendingOrder
+	}
+
+	var limit int64 = 1000
+	var page int64 = 0
+	var offset int64 = 0
+	if in.Pagination != nil {
```

Comment:
Grpc guarantees it won't be nil. If its not specified in the request, it will contain default values. For ints that's 0